tests/virt_autotest/update_package.pm

```diff
@@ -48,8 +48,8 @@ sub update_package {
 sub run {
     my $self = shift;
     #workaroud: skip update package for registered aarch64 tests and because there are conflicts on sles15sp2 XEN
-    $self->update_package() unless (is_registered_sles && is_aarch64);
-    unless ((is_registered_sles && is_aarch64) || is_s390x) {
+    $self->update_package() unless (!!get_var('AUTOYAST') || is_registered_sles && is_aarch64);
+    unless (!!get_var('AUTOYAST') || (is_registered_sles && is_aarch64) || is_s390x) {
```

Comment:
After this change, there will be no grub.cfg uploading anymore. Switching from step-by-step to autoyast install already leads to no grub.cfg/grub default uploading, it seems that they will be uploaded in ```update_pacakage.pm```. At last, even ```update_package.pm``` does not do this. I think this is a big gap compared to previous functionality.

Reply:
The easiest way to solve the problem is as below I think:
```
unless (!!get_var('AUTOYAST') || (is_registered_sles && is_aarch64) || is_s390x) {
    xxxxxx
}
else {
    upload_logs("grub.cfg and grub default file");
}
```

Reply:
@tonyyuan1 Please also help update this part. And use failok=> 1 with upload_logs to make it safe.

Reply:
This is out of scope of this ticket. I will do this time. Please try not to ask to do this in the future.

Reply:
It falls into this pull request, because the original grub config uploading happens in ```set_grub_on_vh('', '', 'xen') ``` which is skipped after this change. It should be kept, because we already skip the first time grub menu needle matching.